hipposerve/web/__init__.py

```diff
@@ -299,8 +324,56 @@ async def logout(request: Request) -> RedirectResponse:
     return new_response
 
 
+@web_router.get("/user/apikey")
+async def read_apikey(request: Request, user: PotentialLoggedInUser):
+    if user is None:
+        new_response = RedirectResponse(
+            url=request.url.path.replace("/user/apikey", "/login"), status_code=302
+        )
+        return new_response
```

Comment:
For stuff like this why not just require a LoggedInUser and we can add an exception handler to the main app to redirect them?

Reply:
See: https://fastapi.tiangolo.com/tutorial/handling-errors/